internal/resource/http.go

```diff
@@ -154,10 +155,29 @@ func NewHttpClient(logger *log.Logger, fetcher Fetcher, timeouts types.Timeouts,
 			},
 		}).Dial,
 		TLSHandshakeTimeout: 10 * time.Second,
-		TLSClientConfig:     &tls.Config{RootCAs: pool},
 	}
+}
 
-	return client, nil
+func (f *Fetcher) newHttpClient() {
+	transport := &http.Transport{
```

Comment:
Why build this manually instead of calling the `newHttpTransport` function?

Reply:
The `newHttpTransport` function was supposed to be deleted. This should be the only place a transport is getting constructed, hence why it's being done manually.